paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentConfirmationOptionKtx.kt

```diff
@@ -0,0 +1,25 @@
+package com.stripe.android.paymentsheet
+
+import com.stripe.android.paymentsheet.model.PaymentSelection
+
+internal fun PaymentSelection.toPaymentConfirmationOption(): PaymentConfirmationOption? {
+    return when (this) {
+        is PaymentSelection.Saved -> PaymentConfirmationOption.Saved(
+            paymentMethod = paymentMethod,
+            optionsParams = paymentMethodOptionsParams,
+        )
+        is PaymentSelection.ExternalPaymentMethod -> PaymentConfirmationOption.ExternalPaymentMethod(
+            type = type,
+            billingDetails = billingDetails,
+        )
+        is PaymentSelection.New -> {
+            PaymentConfirmationOption.New(
+                createParams = paymentMethodCreateParams,
+                optionsParams = paymentMethodOptionsParams,
+                shouldSave = customerRequestedSave == PaymentSelection.CustomerRequestedSave.RequestReuse,
+            )
+        }
+        is PaymentSelection.Link,
+        is PaymentSelection.GooglePay -> null
```

Comment:
not needed for this PR but maybe for a follow up --

how do we handle the payment option confirmation being null? Do we just silently do nothing or do we have analytics or anything?